src/components/sortableList/SortableList.tsx

```diff
@@ -0,0 +1,96 @@
+import React, {useState} from 'react';
+import {LayoutChangeEvent, StyleProp, ViewStyle, FlatList} from 'react-native';
+import Animated, {useAnimatedRef, useAnimatedScrollHandler, useSharedValue} from 'react-native-reanimated';
+import Item from './Item';
+
+const AnimatedFlatList = Animated.createAnimatedComponent(FlatList);
+
+export interface SortableListProps<T> {
+  /**
+   * Plain array of all items
+   */
+  items: Array<T>;
+  /**
+   * Takes an item from items list and renders it into the view
+   */
+  renderItem: ({item, index}: { item: T, index: number }) => JSX.Element;
+  /**
+   * Height of each item (mandatory)
+   */
+  itemHeight: number;
+  /**
+   * Called when the animation of the sort action is finished
+   */
+  onOrderChange: (items: Array<T>) => void;
+  /**
+   * [optional] The size (in pixels) of the draggable area. Enables scrolling when set as it default to the whole width of the item component. 
+   */
+  draggableAreaSize?: number;
+  /**
+   * [optional] The side of the draggable area. defaults to left.
+   */
+  draggableAreaSide?: 'left' | 'right';
+  /**
+   * Styling for the external list View
+   */
+  style?: StyleProp<ViewStyle>;
+  /**
+   * Styling for the internal list View
+   */
+  contentContainerStyle?: StyleProp<ViewStyle>;
+  /**
+   * Used to locate this view in end-to-end tests
+   */
+  testID?: string;
+}
+
+const SortableList = ({items, itemHeight, onOrderChange, renderItem, draggableAreaSize, draggableAreaSide = 'left', style = {}, contentContainerStyle = {}, testID}: SortableListProps<any>) => {
+  const [containerHeight, setContainerHeight] = useState<number>();
+  const positions = useSharedValue({...items.map((_item: any, index: number) => index)});
+  const scrollY = useSharedValue(0);
+  const listRef = useAnimatedRef<FlatList>();
+  const contentHeight = items.length * itemHeight;
```

Comment:
But content size will not change as all items have `position: 'absolute'`

Reply:
Ok, I missed that. 
Is it possible to avoid rendering the items with absolute position? 

What if instead of changing the top position of each item, you use `transform` and change the `translateY` of each item? 
This way the items will stay static and you'll have an actual content size. 
Also, this might help with supporting custom item height sizes